[[Bug 20515]] Segmented control widget highlights off

bwmilby opened the pull request:
`hintLower` handler should not subtract 0.5 from pBound.  This was resulting in an un-highlighted line one pixel wide at the right/bottom side of a cell depending on how the rounding went.

bwmilby committed:
[[Bug 20515]] Segmented control widget highlights off

`hintLower` handler should not subtract 0.5 from pBound.  This was resulting in an un-highlighted line one pixel wide at the right/bottom side of a cell depending on how the rounding went.

bwmilby commented:
Should the bugfix note go in the widget folder itself?  The notes there look a little different than the ones in the main LiveCode repo.

bwmilby committed:
[[Bug 20515]] Add note

Prevent segmented control fill from ending one pixel to the left/bottom of segment.

montegoulding commented on extensions/widgets/segmented/segmented.lcb:
```diff
@@ -379,7 +379,7 @@ end handler
 -- the input span is always fully contained within the output span.
 handler hintLower(in pBound as Number) returns Number
-	return the floor of (pBound - 0.5)
+	return the floor of (pBound)
```
@bwmilby Hmm... perhaps it's meant to be `the ceiling of (pBound - 0.5)` ? I don't think the patch changes the handler to match the comment.

bwmilby replied:
I can't see any visual difference between the two options, but that would be to opposite of the hintUpper handler.

bwmilby committed:
[[Bug 20515]] Segmented control widget highlights off

Changed the code to match the form of the hintUpper handler.  Switched to use ceiling per suggestion from @montegoulding
I did not see any visual difference in the rendering between the two options.

livecodeali commented:
@livecode-vulcan review ok aca5daa

livecode-vulcan commented:
💙 review by @livecodeali ok aca5daa05546e25e54728288c376d13892b5ee0a

livecode-vulcan commented:
😎 test success aca5daa05546e25e54728288c376d13892b5ee0a

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/932)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/948)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/922)
* try-community-universal-ios-iphoneos11.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.0/builds/97)
* try-community-universal-ios-iphonesimulator11.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.0/builds/98)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/926)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/948)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/932)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/948)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/931)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/941)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/963)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/927)
